Forge.TreeWalker/src/TreeWalkerSession.cs

```diff
@@ -534,9 +534,17 @@ internal async Task<string> SelectChild(TreeNode treeNode)
                 {
                     return cs.Child;
                 }
-                if ((bool)await this.EvaluateDynamicProperty(cs.ShouldSelect, typeof(bool)).ConfigureAwait(false))
+                var evaluationResult = await EvaluateDynamicProperty(cs.ShouldSelect, typeof(bool)).ConfigureAwait(false);
+                try
                 {
-                    return cs.Child;
+                    if ((bool)evaluationResult)
+                    {
+                        return cs.Child;
+                    }
+                }
+                catch (Exception e)
+                {
+                    throw new Exception($"An error occurred while casting 'ShouldSelect' expression: '{cs.ShouldSelect}' in the ChildSelector with child '{cs.Child}': '{e.Message}'.", e);
```

Comment:
$"An error occurred while casting ShouldSelect expression: '{cs.ShouldSelect}' in the ChildSelector with child: '{cs.Child}'. Exception: {e.Message}."